engine/access/ingestion/collection_syncer.go

```diff
@@ -390,7 +393,18 @@ func (s *CollectionSyncer) OnCollectionDownloaded(_ flow.Identifier, entity flow
 		return
 	}
 
-	err := indexer.IndexCollection(collection, s.collections, s.transactions, s.logger, s.collectionExecutedMetric)
+	// Create a lock context for indexing
+	lctx := s.lockManager.NewContext()
+	defer lctx.Release()
+
+	err := lctx.AcquireLock(storage.LockInsertCollection)
+	if err != nil {
+		// TODO(leo): should be using irrecoverable.Context
```

Comment:
fatal is OK here. The [new version](engine/access/ingestion2/collection_syncer.go) does use the context, and this code will be removed once we switch to the new engine.